popup.js

```diff
@@ -1,67 +1,176 @@
 }
 
-function get(callback){
-  /* (null) -> (int)
-   *
-   * Gets the amount of overall exp.
+function getExp(skillName, callback){
+  /* Gets exp of a skill from storage.
    */
-
-  chrome.storage.sync.get([storage_id], function (result) {
-    var overall_exp;
-    if (storage_id in result) {
-      overall_exp = result[storage_id];
+  chrome.storage.sync.get([skillName], function (result) {
+    var overallExp;
+    if (skillName in result) {
+      overallExp = result[skillName];
     }
     else{
-      overall_exp = 0;
+      overallExp = 0;
     }
-    callback(overall_exp);
+    callback(overallExp);
   });
 }
 
-function displayExp(exp_value) {
-  $(".expValue").html(exp_value);
-}
-
-function pressButton(){
-  extension_log("Nacisnalem button.");
-  send(parseInt($(".add_value_form").val()),
-       displayExp);
+function displayExp(skillNr) {
+  /* Sets display of a skill with certain number.
+   */
+  getExp(allSkills[skillNr], function (exp) {
+    $(".exp" + skillNr).html(exp);
+  });
 }
 
-function setExp(){
-  get(displayExp);
+function increaseValue(skillNr){
+  var addedExp = $("#add_value_num" + skillNr).val();
+  updateExp(parseInt(addedExp), skillNr, displayExp);
 }
 
 function extension_log (message) {
+  /* Writes to console argument.
+   */
   var script = 'console.log(`' + message + '`);';
   chrome.tabs.executeScript({
     code: script
   });
 }
 
+function createRowTable(skill) {
+  /* Receiving parameters of a skill, creates table row.
+   */
+  var htmlCode = (`
+    <h4 class="skill_name"> ` + skill.skillName + `: </h4>
+    <a class="exp` + skill.skillNr + `"> ` + skill.expValue + ` </a>
+      <div>
+      <input id="add_value_num` + skill.skillNr + `" class="add_value_nums" type="number" name="addValue" value ="">
+      <button id="add_value_button` + skill.skillNr + `" class="add_value_buttons" type="button">Dodaj</button>
+    </div>
+  `);
+  $('#skills').append(htmlCode);
+  // extension_log(htmlCode);
+}
+
+function displayTable (callback) {
+  /* Creates table of skills by adding every skill row by row
+   * and sets display of this skill.
+   */
+  for(i = 0; i < allSkills.length; i++){
+    var skill = {
+      skillName: allSkills[i],
+      expValue: -1, // This value will be updated in the display function.
+      skillNr: i,
+    };
+    createRowTable(skill);
+    displayExp(i);
+  }
+  callback();
+}
+
+function newSkillToTable (nr) {
+  /* Inserts new skill to HTML table of skills.
+   */
+  var skill = {
+    skillName: allSkills[nr],
+    expValue: 0,
+    skillNr: nr,
+  };
+  displayExp(nr);
+  createRowTable(skill);
+}
+
+function addSkill () {
+  /* Adds skill to storage and to current table.
+   */
+  var skillName = $('#skill_name').val();
+  var object = {}
+  object[skillName] = 0;
+  allSkills.push(skillName);
+  chrome.storage.sync.set({skillsArrayId: allSkills});
+  chrome.storage.sync.set(object);
+
+  var logResult = function (result) {
```

Comment:
The code from this line to 128 is only for debug. I think you should add it to the separated function. They are not part of the core code of the function. This function adding skill. That code is not adding skill but testing it was added. For that reason this function is too long.